sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -76,6 +76,9 @@ private[sql] class JSONOptions(
   // Whether to ignore column of all null values or empty array/struct during schema inference
   val dropFieldIfAllNull = parameters.get("dropFieldIfAllNull").map(_.toBoolean).getOrElse(false)
 
+  // Whether to ignore null fields during json generating
+  val ignoreNullFields = parameters.getOrElse("ignoreNullFields", "true").toBoolean
```

Comment:
SQLConf should also take effect here. How about
```
parameters.get("ignoreNullFields").map(_.toBoolean).getOrElse {
  SQLConf.get. jsonGeneratorIgnoreNullFields
}
```

Reply:
Perfect. Then the code in DataSet.scala is not necessary now.